petab/simplify.py

```diff
@@ -0,0 +1,106 @@
+"""Functionality for simplifying PEtab problems"""
+from math import nan
+
+import pandas as pd
+
+from . import Problem
+from .C import *  # noqa: F403
+from .lint import lint_problem
+
+__all__ = [
+    "remove_nan_measurements",
+    "remove_unused_observables",
+    "remove_unused_conditions",
+    "simplify_problem",
+    "condition_parameters_to_parameter_table",
+]
+
+
+def remove_nan_measurements(problem: Problem):
+    """Drop any measurements that are NaN"""
+    problem.measurement_df = problem.measurement_df[
+        ~problem.measurement_df[MEASUREMENT].isna()
+    ]
+    problem.measurement_df.reset_index(inplace=True, drop=True)
+
+
+def remove_unused_observables(problem: Problem):
+    """Remove observables that have no measurements"""
+    measured_observables = set(problem.measurement_df[OBSERVABLE_ID].unique())
+    problem.observable_df = problem.observable_df[
+        problem.observable_df.index.isin(measured_observables)]
+
+
+def remove_unused_conditions(problem: Problem):
+    """Remove conditions that have no measurements"""
+    measured_conditions = \
+        set(problem.measurement_df[SIMULATION_CONDITION_ID].unique())
+    if PREEQUILIBRATION_CONDITION_ID in problem.measurement_df:
+        measured_conditions |= \
+            set(problem.measurement_df[PREEQUILIBRATION_CONDITION_ID].unique())
+
+    problem.condition_df = problem.condition_df[
+        problem.condition_df.index.isin(measured_conditions)]
+
+
+def simplify_problem(problem: Problem):
+    if lint_problem(problem):
+        raise ValueError("Invalid PEtab problem supplied.")
+
+    remove_unused_observables(problem)
+    remove_unused_conditions(problem)
+    condition_parameters_to_parameter_table(problem)
```

Comment:
Why is `remove_nan_measurements` missing here? Are `nan` measurements sometimes useful?

Reply:
Because nan-measurements are not allowed (anymore). This would fail the initial validation step. I still added it, because it's useful for cleaning some older files, where it can be used outside this function.